// @avito/eslint-config-es6/index.js
module.exports = {
    extends: '@avito/eslint-config-es5',
    parser: 'babel-eslint',
    env: {
        es6: true
    },
    rules: {
        // Possible errors
        quotes: [2, 'single', { allowTemplateLiterals: true }],

        // ECMAScript 6
        'arrow-body-style': 0,
        'arrow-parens': 0,
        'arrow-spacing': 2,
        'constructor-super': 2,
        'generator-star-spacing': 0,
        'no-class-assign': 2,
        'no-confusing-arrow': 0,
        'no-const-assign': 2,
        'no-dupe-class-members': 2,
        'no-duplicate-imports': 0,
        'no-new-symbol': 0,
        'no-restricted-imports': 0,
        'no-this-before-super': 2,
        'no-useless-computed-key': 0,
        'no-useless-constructor': 0,
        'no-useless-rename': 0,
        'no-var': 2,
        'object-shorthand': 0,
        'prefer-arrow-callback': 0,
        'prefer-const': 2,
        'prefer-destructuring': 0,
        'prefer-numeric-literals': 0,
        'prefer-rest-params': 0,
        'prefer-spread': 0,
        'prefer-template': 0,
        'require-yield': 0,
        'rest-spread-spacing': 0,
        'sort-imports': 0,
        'symbol-description': 0,
        'template-curly-spacing': 0,
        'yield-star-spacing': 0
    },
    parserOptions: {
        ecmaFeatures: {
            arrowFunctions: true,
            binaryLiterals: true,
            blockBindings: true,
            classes: true,
            defaultParams: true,
            destructuring: true,
            forOf: true,
            generators: true,
            modules: true,
            objectLiteralComputedProperties: true,
            objectLiteralDuplicateProperties: true,
            objectLiteralShorthandMethods: true,
            objectLiteralShorthandProperties: true,
            octalLiterals: true,
            regexUFlag: true,
            regexYFlag: true,
            restParams: true,
            spread: true,
            superInFunctions: true,
            templateStrings: true,
            unicodeCodePointEscapes: true,
            globalReturn: true,
            experimentalObjectRestSpread: true
        }
    }
};
