// @avito/eslint-config-react/index.js
module.exports = {
    extends: '@avito/eslint-config-es6',
    rules: {
        'jsx-quotes': [2, 'prefer-single'],
        'react/jsx-uses-vars': 2,
        'react/jsx-no-duplicate-props': [2, { 'ignoreCase': true }],
        'react/jsx-no-undef': 2,
        'react/jsx-uses-react': 2,
        'react/jsx-no-bind': [2, { 'ignoreRefs': true, 'allowArrowFunctions': true }],
        'react/jsx-no-target-blank': 2,
        'react/jsx-closing-bracket-location': [2, 'after-props'],
        'react/jsx-curly-spacing': 2,
        'react/jsx-wrap-multilines': 2,
        'react/jsx-equals-spacing': 2,
        'react/jsx-pascal-case': 2,
        'react/jsx-handler-names': 2,
        'react/jsx-key': 2,
        'react/jsx-indent-props': 2,
        'react/jsx-first-prop-new-line': [2, 'multiline'],
        'react/self-closing-comp': [2, { html: false }],
        'react/sort-comp': [2, {
            order: [
                'static-methods',
                'lifecycle',
                'rendering',
                '/^handle.+$/',
                'everything-else'
            ],
            groups: {
                rendering: [
                    'render',
                    '/^render.+$/'
                ]
            }
        }],
        'react/prefer-es6-class': 2,
        'react/prefer-stateless-function': [2, { 'ignorePureComponents': true }],
        'react/no-direct-mutation-state': 2,
        'react/no-multi-comp': [2, { 'ignoreStateless': true }],
        'react/no-string-refs': 2,
        'react/no-danger': 2,
        'react/no-deprecated': 2,
        'react/no-is-mounted': 2,
        'react/no-unknown-property': 2,
        'react/prop-types': 2,
        'react/react-in-jsx-scope': 2
    },
    parserOptions: {
        ecmaFeatures: {
            jsx: true
        }
    },
    plugins: [
        'react'
    ],
    settings: {
        react: {
            pragma: 'React',
            version: '15.4.1'
        }
    }
};
