// @avito/eslint-config-es5/index.js
module.exports = {
    env: {
        browser: true,
        node: true,
        worker: true
    },
    rules: {
        // Possible errors
        'no-await-in-loop': 0,
        'no-cond-assign': 2,
        'no-console': 2,
        'no-constant-condition': 2,
        'no-control-regex': 2,
        'no-debugger': 2,
        'no-dupe-args': 2,
        'no-dupe-keys': 2,
        'no-duplicate-case': 2,
        'no-empty-character-class': 2,
        'no-empty': [2, { allowEmptyCatch: true }],
        'no-ex-assign': 2,
        'no-extra-boolean-cast': 2,
        'no-extra-parens': 0,
        'no-extra-semi': 2,
        'no-func-assign': 2,
        'no-inner-declarations': [2, 'functions'],
        'no-invalid-regexp': 2,
        'no-irregular-whitespace': 2,
        'no-obj-calls': 2,
        'no-prototype-builtins': 0,
        'no-regex-spaces': 2,
        'no-sparse-arrays': 2,
        'no-template-curly-in-string': 0,
        'no-unexpected-multiline': 2,
        'no-unreachable': 2,
        'no-unsafe-finally': 0,
        'no-unsafe-negation': 2,
        'use-isnan': 2,
        'valid-jsdoc': [2, {
            preferType: {
                object: 'Object',
                function: 'Function',
                array: 'Array',
                Number: 'number',
                String: 'string',
                Boolean: 'boolean',
                Null: 'null',
                Symbol: 'symbol',
                Undefined: 'undefined'
            },
            matchDescription: '.+',
            requireReturn: false,
            requireParamDescription: false,
            requireReturnDescription: false
        }],
        'valid-typeof': 2,

        // Best practices
        'accessor-pairs': 0,
        'array-callback-return': 0,
        'block-scoped-var': 2,
        'class-methods-use-this': 0,
        complexity: 0,
        'consistent-return': 2,
        curly: 2,
        'default-case': 0,
        'dot-location': [2, 'property'],
        'dot-notation': 2,
        eqeqeq: 2,
        'guard-for-in': 2,
        'no-alert': 0,
        'no-caller': 2,
        'no-case-declarations': 0,
        'no-div-regex': 2,
        'no-else-return': 0,
        'no-empty-function': 0,
        'no-empty-pattern': 2,
        'no-eq-null': 2,
        'no-eval': 2,
        'no-extend-native': 2,
        'no-extra-bind': 2,
        'no-extra-label': 0,
        'no-fallthrough': 2,
        'no-floating-decimal': 2,
        'no-global-assign': 2,
        'no-implicit-coercion': 2,
        'no-implicit-globals': 0,
        'no-implied-eval': 2,
        'no-invalid-this': 0,
        'no-iterator': 2,
        'no-labels': 2,
        'no-lone-blocks': 2,
        'no-loop-func': 2,
        'no-magic-numbers': 0,
        'no-multi-spaces': 2,
        'no-multi-str': 2,
        'no-new-func': 2,
        'no-new-wrappers': 2,
        'no-new': 0,
        'no-octal': 2,
        'no-octal-escape': 2,
        'no-param-reassign': 0,
        'no-proto': 2,
        'no-redeclare': 2,
        'no-restricted-properties': 0,
        'no-return-assign': 0,
        'no-return-await': 0,
        'no-script-url': 2,
        'no-self-assign': 0,
        'no-self-compare': 2,
        'no-sequences': 2,
        'no-throw-literal': 2,
        'no-unmodified-loop-condition': 0,
        'no-unused-expressions': 2,
        'no-unused-labels': 0,
        'no-useless-call': 2,
        'no-useless-concat': 2,
        'no-useless-escape': 0,
        'no-useless-return': 0,
        'no-void': 2,
        'no-warning-comments': 0,
        'no-with': 2,
        'prefer-promise-reject-errors': 0,
        radix: 2,
        'require-await': 0,
        'vars-on-top': 0,
        'wrap-iife': [2, 'any'],
        yoda: 2,

        // Strict Mode
        strict: 0,

        // Variables
        'init-declarations': 0,
        'no-catch-shadow': 2,
        'no-delete-var': 2,
        'no-label-var': 0,
        'no-restricted-globals': 0,
        'no-shadow-restricted-names': 2,
        'no-shadow': 0,
        'no-undef-init': 2,
        'no-undef': [2, { typeof: true }],
        'no-undefined': 0,
        'no-unused-vars': [2, { vars: 'all', args: 'after-used', varsIgnorePattern: '^React$' }],
        'no-use-before-define': [2, 'nofunc'],

        // Node.js and CommonJS
        'callback-return': 0,
        'global-require': 0,
        'handle-callback-err': 0,
        'no-mixed-requires': 0,
        'no-new-require': 0,
        'no-path-concat': 0,
        'no-process-env': 0,
        'no-process-exit': 0,
        'no-restricted-modules': 0,
        'no-sync': 0,

        // Stylistic Issues
        'array-bracket-spacing': 2,
        'block-spacing': 0,
        'brace-style': 2,
        camelcase: 2,
        'capitalized-comments': 0,
        'comma-dangle': 2,
        'comma-spacing': 2,
        'comma-style': 2,
        'computed-property-spacing': 0,
        'consistent-this': 0,
        'eol-last': 2,
        'func-call-spacing': 0,
        'func-name-matching': 0,
        'func-names': 0,
        'func-style': 0,
        'id-blacklist': 0,
        'id-length': 0,
        'id-match': 0,
        indent: [2, 4, { SwitchCase: 1 }],
        'jsx-quotes': 0,
        'key-spacing': 2,
        'keyword-spacing': 2,
        'line-comment-position': 0,
        'linebreak-style': [2, 'unix'],
        'lines-around-comment': 0,
        'lines-around-directive': 0,
        'max-depth': 0,
        'max-len': [1, { code: 120, ignoreUrls: true }],
        'max-lines': 0,
        'max-nested-callbacks': 0,
        'max-params': 0,
        'max-statements-per-line': 0,
        'max-statements': 0,
        'multiline-ternary': 0,
        'new-cap': [2, { capIsNew: false }],
        'new-parens': 0,
        'newline-after-var': 0,
        'newline-before-return': 0,
        'newline-per-chained-call': 0,
        'no-array-constructor': 0,
        'no-bitwise': 0,
        'no-continue': 0,
        'no-inline-comments': 0,
        'no-lonely-if': 0,
        'no-mixed-operators': 0,
        'no-mixed-spaces-and-tabs': 2,
        'no-multi-assign': 0,
        'no-multiple-empty-lines': [2, { max: 1 }],
        'no-negated-condition': 0,
        'no-nested-ternary': 0,
        'no-new-object': 0,
        'no-plusplus': 0,
        'no-restricted-syntax': 0,
        'no-tabs': 0,
        'no-ternary': 0,
        'no-trailing-spaces': 2,
        'no-underscore-dangle': 2,
        'no-unneeded-ternary': 0,
        'no-whitespace-before-property': 0,
        'object-curly-newline': 0,
        'object-curly-spacing': [2, 'always'],
        'object-property-newline': 0,
        'one-var-declaration-per-line': [2, 'always'],
        'one-var': [2, 'never'],
        'operator-assignment': 0,
        'operator-linebreak': [2, 'after', { overrides: { '&&': 'ignore', '||': 'ignore' } }],
        'padded-blocks': [2, 'never'],
        'quote-props': [2, 'as-needed'],
        quotes: [2, 'single'],
        'require-jsdoc': 1,
        'semi-spacing': 2,
        semi: 2,
        'sort-keys': 0,
        'sort-vars': 0,
        'space-before-blocks': 2,
        'space-before-function-paren': [2, 'never'],
        'space-in-parens': 2,
        'space-infix-ops': 2,
        'space-unary-ops': [2, { words: true, nonwords: false }],
        'spaced-comment': 2,
        'template-tag-spacing': 0,
        'unicode-bom': 0,
        'wrap-regex': 0
    }
};
